src/worker/functions/commit-files.js

```diff
@@ -0,0 +1,109 @@
+const getCurrentCommit = async (octokit, username, repo, branch = 'main') => {
+  const { data: refData } = await octokit.request(
+    `GET /repos/${username}/${repo}/git/ref/heads/${branch}`,
+    {
+      owner: username,
+      repo: repo,
+      ref: `heads/${branch}`,
+    }
+  );
+
+  const commitSha = refData.object.sha;
+  const { data: commitData } = await octokit.request(
+    `GET /repos/${username}/${repo}/git/commits/${commitSha}`,
+    {
+      owner: username,
+      repo: repo,
+      commit_sha: commitSha,
+    }
+  );
+  return {
+    commitSha,
+    treeSha: commitData.tree.sha,
+  };
+};
+
+const createBlobForFile = async (octokit, username, repoName, files) => {
+  let blobDataPromises = [];
+  for (const file of files) {
+    let reader = new FileReader();
+    await reader.readAsArrayBuffer(file);
+    const promise = new Promise((resolve, reject) => {
+      reader.onload = async () => {
+        const imageBuffer = reader.result;
+        const byteArray = new Uint8Array(imageBuffer);
+        const blob = new Blob([byteArray]);
+        const BlobURL = URL.createObjectURL(blob);
+        const blobData = octokit.request(`POST /repos/${username}/${repoName}/git/blobs`, {
```

Comment:
Either you need to `await` this call, or better, you should return it as the Promise to wait on in line 48 below.

Reply:
Will be addressed